lib/reader.js

```diff
@@ -349,6 +349,31 @@ Reader.prototype.close = function(fn){
 };
 
 /**
+ * Close the connections.
+ *
+ * @param {Function} [fn]
+ * @api public
+ */
+
+Reader.prototype.end = function(fn){
+  debug('end');
+
+  clearInterval(this.timer);
+
+  var n = this.conns.size();
```

Comment:
This does not solve the problem, the second `end` call would still call `fn`, and would call it immediately and assume all connections were closed, even if they were still closing.